### ISSUE TITLE
Consider joining fast-pack org?

### ISSUE THREAD

**Issue by nyurik** [zentures/encoding#9]
Recently @lemire, author of the core fastpfor algorithm, transferred all his code to the new github org [fast-pack](https://github.com/fast-pack).  Would it make sense to move this project to that org too, so that all implementations can be kept closer together and maintained by a community?  Current project admins will continue being admins after the transfer, and we could automate publishing to the package registry. Thx!

**Comment by lemire** [comment]
I support this.

**Comment by zhenjl** [comment]
hey folks, sincere apologies. I got this during vacation and forgot about it after. Yes I'd be happy to move it. Please tell me what needs to be done.

**Comment by nyurik** [comment]
I think the simplest way is to add me as an admin to your project, and I will move it, and give you access?

**Comment by zhenjl** [comment]
Done

**Comment by nyurik** [comment]
Thx!  One more thing before i move it - how should we call it (repo name)?   @lemire you might have some thoughts on this...  Some ideas:

* `GoFastPFOR`  (similar to `JavaFastPFOR` which is already there)
* `FastPFOR-Go`
* ...?

**Comment by lemire** [comment]
I recommend using the Go naming practices:

Keep package names short, preferably one word or a very short phrase.
Use lower case for package names (e.g., net/http, encoding/json).

**Comment by nyurik** [comment]
sure, but the github repo name should not confuse people either - i.e. someone looking through the list of repos shouldn't be confused about the content of the `encoding` repo, unless you want to move all languages into a single monorepo? (it would create issues with the releases)

**Comment by lemire** [comment]
@nyurik 

> sure, but the github repo name should not confuse people either

Right. But taking into account that the name will be prefixed by fast-pack.


I think that including 'go' or 'Go' in the name of a Go package goes against the Go naming practice.

Please see the following organization where one package is a Go package:

https://github.com/RoaringBitmap

:-)

**Comment by nyurik** [comment]
I must be getting confused already :)   Is this a requirement for the git *repo* to have the same name as the go *package*?

**Comment by nyurik** [comment]
P.S. Note that in Rust, it is usually the same convention - the git repo may have an `-rs` suffix to indicate repo's content, but the rust **crate** name would not have it

**Comment by lemire** [comment]
@nyurik 

> I must be getting confused already :) Is this a requirement for the git repo to have the same name as the go package?


The typical usage is like so...


```Go
import "github.com/fast-pack/nameofrepo"

...

nameofrepo.FunctionFromRepo
```

You do not include the word 'go' in the repo (usually).

**Comment by lemire** [comment]
(Note that I am not telling anyone how to name it, I am just stating what the usage is.)

**Comment by nyurik** [comment]
@lemire thx for the explanation, makes sense.  Are you ok to preserve the name as encoding in the new org, or should it be called `fastpfor` or something similar?

**Comment by lemire** [comment]
> Are you ok to preserve the name as encoding in the new org, or should it be called fastpfor or something similar?

I am neutral but fastpfor won't work because there is [FastPFor](https://github.com/fast-pack/FastPFor) and I do not think GitHub will allow it.


Grok suggests the following names:

- intpack
- bitpack
- intcomp
- numcrunch
- bitcrunch
- intencode
- numshrink
- effint
- numpack
- compactint
